Assets/Scripts/GameState/LeaderboardData.cs

```diff
@@ -56,58 +75,54 @@ public static void SaveScores(Highscores scores)
         stream.Close();
     }
 
-    public static Highscores LoadScores()
+    /// <summary>
+    /// Load all saved scores
+    /// </summary>
+    /// <returns>Returns a ordered list of Highscore structs</returns>
+    public static List<Highscore> LoadScores()
     {
         if (File.Exists(path))
         {
             BinaryFormatter formatter = new BinaryFormatter();
             FileStream stream = new FileStream(path, FileMode.Open);
 
-            Highscores data = formatter.Deserialize(stream) as Highscores;
-            stream.Close();
+            List<Highscore> data;
 
-            if(data.name.Length < 10 || data.score.Length < 10)
+            // Check if the file is corrupted
+            try
             {
-                // Scores are missing, delete the data and reset
+                data = formatter.Deserialize(stream) as List<Highscore>;
+            }
+            catch
```

Comment:
Having an empty `catch` clause should generally be avoided, as it might hide the fact that unexpected or more serious exceptions are thrown instead of the exception types that we would normally expect:
```suggestion
            } catch (<Some specific exception class>)
```
If you're unable to find any fitting exception classes, you could add just `(Exception)`, which would communicate explicitly that you want to catch all exceptions. 🙂